### DISCUSSION TITLE
How to do metadata for NEAR NFTs?

### DISCUSSION THREAD

**Discussion by 7paul7** [TENK-DAO/tenk#30] [Q&A] [answered]
When I generate my profile pictures how do I generate the metadata? Is it similar to SOL NFTs where you have a corresponding .json file for each JPEG. Im trying to generate my images now if someone has info I would love it!

**Comment by willemneal** [comment]
Yep!  All files end up in a directory numbered `0.jpeg`, `0.json` so that later the IPFS hash leads to the directory and then each image and its metadata can be found with its id, e.g. `0`. @starpause knows more about the metadata generation part.

**Comment by starpause** [comment, marked as answer]
@7paul7 how are you doing the metadata generation? 

For 2d projects we recommend [hashlips](https://github.com/HashLips/hashlips_art_engine) which will also generate the metadata for you. We have a [PR open which adds NEAR support](https://github.com/HashLips/hashlips_art_engine/pull/552) so you could use that branch if you like. Alternately, there are only 2 changes to the metadata keys which needs to happen for the JSON generated by hashlips to match up with the [NEP171 standard](https://nomicon.io/Standards/NonFungibleToken/Metadata.html):

`"name:"` -> `"title:"`
`"image:"` -> `"media:"`
